diff -up avr-gcc-9.2.0/gcc-9.2.0/gcc/config/avr/avr.h.mint8 avr-gcc-9.2.0/gcc-9.2.0/gcc/config/avr/avr.h
--- avr-gcc-9.2.0/gcc-9.2.0/gcc/config/avr/avr.h.mint8	2019-01-01 13:31:55.000000000 +0100
+++ avr-gcc-9.2.0/gcc-9.2.0/gcc/config/avr/avr.h	2019-08-21 10:46:22.765490163 +0200
@@ -149,6 +149,9 @@ FIXME: DRIVER_SELF_SPECS has changed.
 #define SIZE_TYPE (INT_TYPE_SIZE == 8 ? "long unsigned int" : "unsigned int")
 #define PTRDIFF_TYPE (INT_TYPE_SIZE == 8 ? "long int" :"int")
 
+#define CHAR16_TYPE (INT_TYPE_SIZE == 8 ? "long unsigned int" : "unsigned int")
+#define CHAR32_TYPE (INT_TYPE_SIZE == 8 ? "long long unsigned int" : "long unsigned int")
+
 #define WCHAR_TYPE_SIZE 16
 
 #define FIRST_PSEUDO_REGISTER 36
diff -up avr-gcc-9.2.0/gcc-9.2.0/gcc/defaults.h.mint8 avr-gcc-9.2.0/gcc-9.2.0/gcc/defaults.h
--- avr-gcc-9.2.0/gcc-9.2.0/gcc/defaults.h.mint8	2019-08-21 10:46:22.766490159 +0200
+++ avr-gcc-9.2.0/gcc-9.2.0/gcc/defaults.h	2019-08-21 10:50:33.058604195 +0200
@@ -587,17 +587,21 @@ see the files COPYING3 and COPYING.RUNTI
 #define CHAR8_TYPE "unsigned char"
 #endif
 
+#ifndef CHAR16_TYPE
 #ifdef UINT_LEAST16_TYPE
 #define CHAR16_TYPE UINT_LEAST16_TYPE
 #else
 #define CHAR16_TYPE "short unsigned int"
 #endif
+#endif
 
+#ifndef CHAR32_TYPE
 #ifdef UINT_LEAST32_TYPE
 #define CHAR32_TYPE UINT_LEAST32_TYPE
 #else
 #define CHAR32_TYPE "unsigned int"
 #endif
+#endif
 
 #ifndef WCHAR_TYPE
 #define WCHAR_TYPE "int"
