--- avr-gcc-6.3.0/gcc-6.3.0/gcc/ubsan.c	2016-12-07 23:39:29.000000000 +0100
+++ avr-gcc-6.3.0/gcc-6.3.0/gcc/ubsan.c	2017-03-08 18:35:56.000000000 +0100
@@ -1471,7 +1486,7 @@ ubsan_use_new_style_p (location_t loc)
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
