--- gcc/toplev.c.orig	2008-04-24 13:59:01.000000000 -0400
+++ gcc/toplev.c	2009-01-30 22:12:14.000000000 -0500
@@ -541,6 +541,7 @@
 /* Given X, an unsigned number, return the largest int Y such that 2**Y <= X.
    If X is 0, return -1.  */
 
+/*
 int
 floor_log2 (unsigned HOST_WIDE_INT x)
 {
@@ -572,10 +573,12 @@
 
   return t;
 }
+*/
 
 /* Return the logarithm of X, base 2, considering X unsigned,
    if X is a power of 2.  Otherwise, returns -1.  */
 
+/*
 int
 exact_log2 (unsigned HOST_WIDE_INT x)
 {
@@ -590,7 +590,7 @@
   return floor_log2 (x);
 #endif
 }
-
+*/
 #endif /*  GCC_VERSION < 3004 || !defined (__cplusplus)  */
 
 /* Handler for fatal signals, such as SIGSEGV.  These are transformed
--- gcc/toplev.h.orig	2007-09-23 15:18:27.000000000 -0400
+++ gcc/toplev.h	2009-02-13 20:28:02.000000000 -0500
@@ -155,10 +155,16 @@
 extern bool fast_math_flags_set_p	(void);
 
 /* Return log2, or -1 if not exact.  */
-extern int exact_log2                  (unsigned HOST_WIDE_INT);
+// extern int exact_log2                  (unsigned HOST_WIDE_INT);
+
+# if defined(__GNUC_STDC_INLINE__) || defined(__GNUC_GNU_INLINE__)
+#   define MY_EXTERNAL_INLINE_PROTO extern __attribute__ ((__gnu_inline__)) inline
+# else 
+#   define MY_EXTERNAL_INLINE_PROTO extern inline
+# endif
 
 /* Return floor of log2, with -1 for zero.  */
-extern int floor_log2                  (unsigned HOST_WIDE_INT);
+//extern int floor_log2                  (unsigned HOST_WIDE_INT);
 
 /* Inline versions of the above for speed.  */
 #if GCC_VERSION >= 3004
@@ -173,19 +179,19 @@
 #  define CTZ_HWI __builtin_ctz
 # endif
 
-extern inline int
+static inline int
 floor_log2 (unsigned HOST_WIDE_INT x)
 {
   return x ? HOST_BITS_PER_WIDE_INT - 1 - (int) CLZ_HWI (x) : -1;
 }
 
-extern inline int
+static inline int
 exact_log2 (unsigned HOST_WIDE_INT x)
 {
   return x == (x & -x) && x ? (int) CTZ_HWI (x) : -1;
 }
-#endif /* GCC_VERSION >= 3004 */

+#endif /* GCC_VERSION >= 3004 */
 /* Functions used to get and set GCC's notion of in what directory
    compilation was started.  */
 
